fastestimator/util/util.py

```diff
@@ -214,6 +215,38 @@ def _custom_tf_print(*args, **kwargs):
     print_v2(*args, **kwargs)
 
 
+def is_valid_file(file_path: str) -> bool:
+    """Validate whether file is valid or not.
+
+    Args:
+        file_path: location of the input file.
+
+    Returns:
+        Whether the file is valid.
+    """
+    if not os.path.exists(file_path):
+        return False
+    suffix = Path(file_path).suffix
+    try:
+        if suffix == '.zip':
+            import zipfile
+            zip_file = zipfile.ZipFile(file_path)
+            _ = zip_file.namelist()
+        elif suffix == '.gz':
+            if file_path.endswith('.tar.gz'):
+                import tarfile
+                with tarfile.open(file_path) as img_tar:
+                    _ = img_tar.getmembers()
+            else:
+                import gzip
+                f = gzip.open(file_path, 'rb')
+                _ = f.read()
+        return True
+    except Exception as e:
+        print(e)
```

Comment:
printing e here seems pretty messy though? should we just `except Exception` and ignore the codacy complaint?

Reply:
But would deleting the file cause issues with different threads downloading it?